### DISCUSSION TITLE
Spec wording

### DISCUSSION THREAD

**Discussion by jromero** [olivierlacan/keep-a-changelog#507] [Q&A]
In trying to comprehend what the actual format is outside of the hosted example. I think it would help if common spec (aka legal) wording were to be used. eg. SHALL, MUST, etc. 

For example this would be what the guidelines would look like. It would make parsers using this format know exactly what information is optional versus required.

> - There SHOULD be an entry for every single version.
> - The same types of changes MUST be grouped.
> - Versions and sections SHOULD be linkable.
> - The latest version MUST comes first.
> - The release date of each version SHOULD is displayed.
> - Changelogs MUST mention whether the project follows Semantic Versioning.

**Comment by tofu-rocketry** [comment]
The common spec for this is [RFC 2119 - Key words for use in RFCs to Indicate Requirement Levels](https://tools.ietf.org/html/rfc2119)

**Comment by olivierlacan** [comment]
I've stated several times that I have no interest in making this project anything more that a set of guidelines to help people reason about changelogs and release communication surrounding their projects better.

Legalese, like machine-oriented language, is the opposite direction I'm walking toward. I'm interested in human-friendliness, in recommendations that aren't edicts but pieces of hopefully relevant advice that can clearly and obviously benefit any project.

Using strongly coercive language is not something I think is necessary, it sort of eradicates any sense of these guidelines being adaptable, flexible, and debatable in this forum. If you think that *you* need strictness, then you can feel free to enforce it. I'm not going to go in that direction. 

Does that make life harder for parsers? Absolutely. But parsers are machine language, they can do the work to support edge cases and alternative uses. Or they can find another source of truth than the human-focused changelog. A source that is also used a reference to construct the changelog output for humans — after editorial review by project maintainers and contributors. 

I am however interested in standardizing the formatting of changelogs as much as possible. If that can have the fringe benefit of also making life easier for parsers, great. 😃
